src/common/options.cc

```diff
@@ -6462,6 +6462,22 @@ std::vector<Option> get_rgw_options() {
     .set_description("Max number of keys to remove from garbage collector log in a single operation")
     .add_see_also({"rgw_gc_max_objs", "rgw_gc_obj_min_wait", "rgw_gc_processor_max_time", "rgw_gc_max_concurrent_io"}),
 
+    Option("rgw_gc_max_urgent_data_size", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
+    .set_default(3072)
+    .set_description("maximum allowed size of urgent data in queue head for gc"),
+
+    Option("rgw_gc_max_queue_size", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
+    .set_default(128_M)
```

Comment:
does `rgw_gc_max_queue_size` include the size of the head, or can the actual object size grow to `max_head_size + rgw_gc_max_queue_size`? if so, we could get osd errors because of `osd_max_object_size` before we actually fill the cls_queue

Reply:
oops, sorry yes rgw_gc_max_queue_size should have taken into account the head size also, which means i should subtract the head size from the osd max object size to arrive at rgw gc max queue size.